scripts/storybook/src/utils.js

```diff
@@ -210,34 +210,57 @@ function getImportMappingsForExportToSandboxAddon(allPackageInfo = getAllPackage
  *
  * This helper is useful for creating aggregated storybooks which will generate multiple stories across packages.
  *
- * @param {{packageName:string,callerPath:string}} options
+ * @param {{packageName:string,callerPath:string,excludeStoriesInsertionFromPackages?:string[]}} options
  * @returns
  */
 function getPackageStoriesGlob(options) {
-  const projectMetadata = getProjectMetadata(options.packageName);
+  const projects = getAllProjects();
+
+  const excludeStoriesInsertionFromPackages = options.excludeStoriesInsertionFromPackages ?? [];
+  const projectMetadata = getMetadata(options.packageName, projects);
 
   /** @type {{name:string;version:string;dependencies?:Record<string,string>}} */
   const packageJson = JSON.parse(
     fs.readFileSync(path.resolve(workspaceRoot, projectMetadata.root, 'package.json'), 'utf-8'),
   );
 
-  packageJson.dependencies = packageJson.dependencies ?? {};
-
-  const dependencies = { ...packageJson.dependencies };
+  const dependencies = { ...(packageJson.dependencies ?? {}) };
   const rootOffset = offsetFromRoot(options.callerPath.replace(workspaceRoot, ''));
+  const packages = Object.keys(dependencies);
 
-  return Object.keys(dependencies)
-    .filter(pkgName => pkgName.startsWith('@fluentui/'))
+  const result = packages
+    .filter(pkgName => pkgName.startsWith('@fluentui/') && !excludeStoriesInsertionFromPackages.includes(pkgName))
     .map(pkgName => {
       const storiesGlob = '**/@(index.stories.@(ts|tsx)|*.stories.mdx)';
-      const pkgMetadata = getProjectMetadata(pkgName);
+      const pkgMetadata = getMetadata(pkgName, projects);
 
       if (fs.existsSync(path.resolve(workspaceRoot, pkgMetadata.root, 'stories'))) {
         return `${rootOffset}${pkgMetadata.root}/stories/${storiesGlob}`;
       }
 
       return `${rootOffset}${pkgMetadata.root}/src/${storiesGlob}`;
     });
+
+  return result;
+
+  // ========================================
+  // Helpers
+  // ========================================
+
+  function getAllProjects() {
+    const tree = new FsTree(workspaceRoot, false);
+    return getProjects(tree);
```

Comment:
this is the main perf boost, using readProjectConfiguration for many projects causes huge performance overhead